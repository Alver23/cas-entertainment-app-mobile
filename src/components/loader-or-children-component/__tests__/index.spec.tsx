// Dependencies
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Text } from 'react-native';

// Components
import Loading from '@components/loading';

// Under test file
import LoaderOrChildrenComponent from '../index';

describe('<LoaderOrChildrenComponent />', () => {
  let component: ShallowWrapper;

  const props = {
    loading: false,
    children: <Text />,
  };

  beforeEach(() => {
    component = shallow(<LoaderOrChildrenComponent {...props} />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correctly when loading is false', () => {
    expect(component.find(Text).exists()).toBeTruthy();
  });

  it('should render correctly when loading is true', () => {
    component.setProps({ ...props, loading: true });
    expect(component.find(Loading).exists()).toBeTruthy();
  });
});
