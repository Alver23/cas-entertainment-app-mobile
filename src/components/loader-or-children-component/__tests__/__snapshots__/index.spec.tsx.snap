// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoaderOrChildrenComponent /> should save a snapshot of the component 1`] = `
<Fragment>
  <Text />
</Fragment>
`;
